### PR TITLE
test: 🧪 Controller and Service

### PR DESCRIPTION
**Pull request by RobertDataWow**
Create RSpec Controller test for Book, Review model.

Create a mock TestService class just to write a test for it.

### CODE REVIEW

File: spec/services/test_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe TestService, type: :service do
+  describe '.hello' do
+    subject { described_class }
+
+    context 'no arg' do
```
Comment (komsun-k) [inline spec/services/test_service_spec.rb]:
```suggestion
    context 'without arg' do
```

File: spec/services/test_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      it 'return default' do
```
Comment (komsun-k) [inline spec/services/test_service_spec.rb]:
```suggestion
      it 'returns default' do
```

File: spec/services/test_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        expect(subject.test_return).to eq('default')
+      end
+    end
+
+    context 'have arg' do
+      it 'return arg' do
```
Comment (komsun-k) [inline spec/services/test_service_spec.rb]:
```suggestion
      it 'returns arg' do
```

File: spec/controllers/book_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      context 'book not exist' do
+        let(:params) { { id: -1 } }
+        it 'raise error not found' do
+          expect { subject }.to raise_error(ActiveRecord::RecordNotFound)
+        end
+      end
+    end
+
+    context 'when user not sign in' do
+      it 'unauthorize access' do
```
Comment (komsun-k) [inline spec/controllers/book_controller_spec.rb]:
```suggestion
      it 'unauthorizes access' do
```

File: spec/controllers/book_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      context 'book exist' do
+        it 'returns book' do
+          expect(subject.status).to eq(200)
+          expect(assigns(:book)).to eq(Book.find(book.id))
+        end
+      end
+
+      context 'book not exist' do
+        let(:params) { { id: -1 } }
+        it 'raise error not found' do
```
Comment (komsun-k) [inline spec/controllers/book_controller_spec.rb]:
```suggestion
        it 'raises error not found' do
```

File: spec/controllers/book_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      before { sign_in user }
+
+      it 'returns all books' do
+        expect(subject.status).to eq(200)
+        expect(assigns(:books)).to eq(Book.all)
+      end
+    end
+
+    context 'when user not sign in' do
+      it 'unauthorize access' do
```
Comment (komsun-k) [inline spec/controllers/book_controller_spec.rb]:
```suggestion
      it 'unauthorizes access' do
```

**Review by komsun-k — APPROVED** [review]

**Comment by komsun-k** [comment]
Test suit is good, just wording on some `it` blocks.

File: app/services/test_service.rb
Comment (WatcharitNew) [inline app/services/test_service.rb:1]: if file not use, please remove it

**Review by WatcharitNew — APPROVED** [review]

**Pull request merged**